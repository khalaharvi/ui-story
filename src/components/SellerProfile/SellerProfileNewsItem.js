import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Image from 'components/Image';

import { cx } from './SellerProfile';

const SellerProfileNewsItem = ({ news }) => (
  <Fragment>
    <Image src={news.image} alt={news.title} className={cx('news-image')} />
    <i className={cx('news-type')}>{news.type}</i><br />
    <b className={cx('news-title')}>{news.title}</b>
  </Fragment>
);

SellerProfileNewsItem.propTypes = {
  news: PropTypes.shape({
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
  }).isRequired
};

export default SellerProfileNewsItem;
