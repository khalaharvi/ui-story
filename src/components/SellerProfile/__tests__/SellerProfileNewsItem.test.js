import React from 'react';
import { shallow } from 'enzyme';

import SellerProfileNewsItem from '../SellerProfileNewsItem';

describe('<SellerProfileNewsItem />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SellerProfileNewsItem news={{
          id: 1,
          image: 'https://cdn.deringhall.com/images/681/original/cover.jpg?auto=format&fit=max&q=60&w=670',
          title: 'Remote Possibilities',
          type: 'Press'
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
