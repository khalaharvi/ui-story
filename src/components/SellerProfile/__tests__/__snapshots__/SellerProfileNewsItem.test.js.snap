// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerProfileNewsItem /> renders correctly 1`] = `
<React.Fragment>
  <Image
    alt="Remote Possibilities"
    className="dh-seller-profile-news-image"
    src="https://cdn.deringhall.com/images/681/original/cover.jpg?auto=format&fit=max&q=60&w=670"
  />
  <i
    className="dh-seller-profile-news-type"
  >
    Press
  </i>
  <br />
  <b
    className="dh-seller-profile-news-title"
  >
    Remote Possibilities
  </b>
</React.Fragment>
`;
